### PR TITLE
Model.py for list

### PR DESCRIPTION
**Pull request by qijinze911**
I add three class methods for list API in model.py.
And add tests for those class methods.
Here is the test result:

```
Test Cases for Recommendations 
- Create a recommendation and add it to the database
- Create a recommendation and assert that it exists
- Delete a Recommendation
- Test deserialization of a Recommendation
- Test deserialization of bad data
- Test deserialization of a Recommendation with missing data
- Test deserialization of no data
- Find Recommendation by customer_id
- Find Recommendation by product_id
- Find Recommendation by recommend_type
- Test serialization of a Recommendation
- Update a Recommendation

Name                  Stmts   Miss  Cover
-----------------------------------------
service/__init__.py      11      0   100%
service/models.py        55      0   100%
-----------------------------------------
TOTAL                    66      0   100%
----------------------------------------------------------------------
Ran 12 tests in 0.942s

OK

```

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -113,4 +113,31 @@ def init_db(cls, app):
     def all(cls):
         """ Returns all of the Recommendations in the database """
         cls.logger.info('Processing all Recommendations')
-        return cls.query.all()
\ No newline at end of file
+        return cls.query.all()
+
+    #@classmethod
```
Comment (cheryl1223) [inline service/models.py]: you should remove commented codes from PR to keep the code base clean.

File: service/models.py [hunk 2]
```diff
@@ -113,4 +113,31 @@ def init_db(cls, app):
...
+        return cls.query.filter(cls.product_id == product_id)
+        
+    @classmethod
+    def find_by_recommend_type(cls, recommend_type):
+        """ Returns all of the Recommendation in an recommend_type
+        Args:
+            recommend_type (string): the recommend_type of the Recommendation you want to match
+        """
+        cls.logger.info('Processing recommend_type query for %s ...', recommend_type)
+        return cls.query.filter(cls.recommend_type == recommend_type)
```
Comment (cheryl1223) [inline service/models.py:143]: I can't see find_by_customer_id() method, did you miss it?

**Review by cheryl1223 — COMMENTED** [review]
overall a good job! please address my comments so I can merge it

File: service/models.py (hunk 1 above)
Comment (qijinze911) [reply service/models.py]: That's my fault, I have deleted it.

File: service/models.py (hunk 2 above)
Comment (qijinze911) [reply service/models.py:143]: It shows on my computer but miss on git, and I git push again, now it shows on git.

**Review by cheryl1223 — APPROVED** [review]

**Pull request merged**